### PR TITLE
adding type safety

### PR DESCRIPTION
**Pull request by javabeanz**
a first step for improvement is to use java type safe collections

### CODE REVIEW

File: src/main/java/org/apache/maven/plugin/gpg/AbstractGpgSigner.java [hunk 1]
```diff
@@ -193,9 +195,9 @@ public File generateSignatureForArtifact( File file )
             signature = new File( signatureDirectory, file.getName() + SIGNATURE_EXTENSION );
         }
 
-        if ( signature.exists() )
+        if ( signature.exists() && signature.delete() )
```
Comment (eolivelli) [inline src/main/java/org/apache/maven/plugin/gpg/AbstractGpgSigner.java]: This seems unrelated to the description.
Can you create a soste separate patch?

Comment (javabeanz) [reply src/main/java/org/apache/maven/plugin/gpg/AbstractGpgSigner.java]: I also have added some changes to reduce complexity, as suggested by SonarQube. Some methods have very high complexity, those are still under development. What is "a soste separate patch" ?

Comment (eolivelli) [reply src/main/java/org/apache/maven/plugin/gpg/AbstractGpgSigner.java]: Sorry, 'soste' was a typo, due to auto completion of my browser.

We should keep a patch focused on only just one issue: reformat code, fix a bug....

Overall your patch is more a code clean up. But in this line you are changing the behavior, so I am asking you to file a separate issue

**Comment by rfscholte** [comment]
I've cherrypicked some changes.

**Comment by javabeanz** [comment]
thanks, I hope this was of help. Using Eclipse, non-typesafe code are shown as warnings.
